x-pack/plugins/enterprise_search/public/applications/app_search/components/analytics/analytics_logic.test.ts

```diff
@@ -28,6 +19,11 @@ jest.mock('../engine', () => ({
 import { AnalyticsLogic } from './';
 
 describe('AnalyticsLogic', () => {
+  const { mount } = new LogicMounter(AnalyticsLogic);
+  const { history } = mockKibanaValues;
+  const { http } = mockHttpValues;
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
```

Comment:
I decided to reorganize our tests a bit in the light of all these new fancy helpers. Here's the basic architecture I'm thinking:


```js
/**
 * License header
 */

// Mocks go here
import { LogicMounter, mockHttpValues } from '../../__mocks__';

// 3rd party imports go here
import React from 'react';
import { shallow } from 'enzyme';

// 1st party imports go here
import { SomeComponent } from '../';
import { SomeLogic } from './';

describe('SomeLogic', () => {
  // Destructuring from mocks go here - e.g., `mount` from LogicMounter
  const { mount, unmount } = new LogicMounter(SomeLogic);
  const { http } = mockHttpValues;

  // Constants (e.g.) values, actions go here
  const DEFAULT_VALUES = { ... };
  const foo = 'bar';

  // Jest before/after/all blocks
  beforeEach(() => {
    // ...
  });

  // Start the tests
  it('does something', () => {});
});
```

LMK what you think!

Reply:
I like this, let's make sure it's in a README file somewhere in this codebase.

Reply:
Hm you think so? Maybe I'm overly optimistic, but I'm hoping that we follow this by osmosis - I figure someone opening the codebase for the first time will be like "Oh all the unit tests are written this way, I'll just follow it/copy this file".

My only hesitation with putting this in our README is if it falls out of date, etc. That being said there's definitely a testing section in our main README, so if other folks are +1 to this I can definitely add it there!

Reply:
I think I agree with ya there constance.